src/tqec/compile/specs/library/base.py

```diff
@@ -0,0 +1,485 @@
+            return self._get_temporal_hadamard_pipe_block(spec)
+        # Else, it is a regular temporal junction
+        return self._get_temporal_non_hadamard_pipe_block(spec)
+
+    def _get_temporal_non_hadamard_pipe_block(self, spec: PipeSpec) -> Block:
+        """Returns the block to implement a regular temporal junction
+        without Hadamard transition.
+
+        Args:
+            spec: description of the pipe that should be implemented by this
+                method. Should be a regular (i.e., non-Hadamard) temporal pipe.
+
+        Raises:
+            AssertionError: if the provided ``pipe`` is not a temporal pipe, or
+                if it contains a Hadamard transition.
+
+        Returns:
+            the block to implement the provided
+            ``spec``.
+        """
+        assert spec.pipe_kind.is_temporal
+        assert not spec.pipe_kind.has_hadamard
+
+        z_observable_orientation = (
+            ZObservableOrientation.HORIZONTAL
+            if spec.pipe_kind.x == Basis.Z
+            else ZObservableOrientation.VERTICAL
+        )
+        memory_plaquettes = get_memory_qubit_plaquettes(z_observable_orientation)
+        template = get_memory_qubit_raw_template()
+        return Block([PlaquetteLayer(template, memory_plaquettes) for _ in range(2)])
+
+    def _get_temporal_hadamard_pipe_block(self, spec: PipeSpec) -> Block:
+        """Returns the block to implement a
+        Hadamard temporal junction.
+
+        Note:
+            This method performs the Hadamard transition at the end of the
+            layer that appear first (i.e., temporally before the other, or in
+            other words the one with a lower Z index).
+
+        Args:
+            spec: description of the pipe that should be implemented by this
+                method. Should be a Hadamard temporal pipe.
+
+        Raises:
+            AssertionError: if the provided ``pipe`` is not a temporal pipe, or
+                if it is not a Hadamard transition.
+
+        Returns:
+            the block to implement the provided
+            ``spec``.
+        """
+        assert spec.pipe_kind.is_temporal
+        assert spec.pipe_kind.has_hadamard
+
+        #
+        x_axis_basis_at_head = spec.pipe_kind.get_basis_along(
+            Direction3D.X, at_head=True
+        )
+        assert (
+            x_axis_basis_at_head is not None
+        ), "A temporal pipe should have a non-None basis on the X-axis."
+
+        orientations = (
+            (ZObservableOrientation.HORIZONTAL, ZObservableOrientation.VERTICAL)
+            if x_axis_basis_at_head == Basis.Z
+            else (ZObservableOrientation.VERTICAL, ZObservableOrientation.HORIZONTAL)
+        )
+        first_layer_orientation, second_layer_orientation = orientations
+        hadamard_plaquettes = get_temporal_hadamard_plaquettes(first_layer_orientation)
+        memory_plaquettes = get_memory_qubit_plaquettes(second_layer_orientation)
```

Comment:
Should rather use the `*_rpng_descriptions` variant along with `_get_plaquettes` to use user-provided translator/compiler.